### ISSUE TITLE
Defining an application menu causes panic on Windows

### ISSUE THREAD

**Issue by selvinortiz** [wailsapp/wails#1340]
### Description

# TL;DR
When running `wails dev` on Windows (Parallels Desktop) for any Wails application that uses `*menu.Menu` causes a panic.

## Output from `wails dev`

```panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xba1f93]

goroutine 1 [running, locked to thread]:
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.processMenu(0xc0002189c0?, 0xc000004cf0)
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/internal/frontend/desktop/windows/menu.go:51 +0x93     
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Window).SetApplicationMenu(...)
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/internal/frontend/desktop/windows/menu.go:44
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.NewWindow({0x0?, 0x0}, 0xc00018a0e0)
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/internal/frontend/desktop/windows/window.go:88 +0x393  
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Frontend).Run(0xc000190380, {0xe9f398, 0xc0001d16e0})
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/internal/frontend/desktop/windows/frontend.go:102 +0x78
github.com/wailsapp/wails/v2/internal/frontend/devserver.(*DevWebServer).Run(0xc00018e700, {0xe9f398?, 0xc0001d16e0})
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/internal/frontend/devserver/devserver.go:140 +0x629    
github.com/wailsapp/wails/v2/internal/appng.(*App).Run(0xc00005c3c0)
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/internal/appng/app_dev.go:47 +0x31
github.com/wailsapp/wails/v2.Run(0xd61140?)
        C:/Users/selvinortiz/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.34/wails.go:24 +0x49
main.main()
        C:/Users/selvinortiz/dev/sample/main.go:31 +0x52a
```
  
## New To Wails
> I'm new to wails, so if I'm missing something or I'm doing something wrong, please let me know.

### To Reproduce

1. Initialize a project with `wails init -n sample`
2. Set `Menu` in `main.go` to something other than `nil`
3. Run `wails dev`

### The following `main.go` file has a simple menu that uses the built-in `menu.AppMenu()`

```
package main

import (
	"embed"
	"log"

	"github.com/wailsapp/wails/v2/pkg/menu"
	"github.com/wailsapp/wails/v2/pkg/options/mac"

	"github.com/wailsapp/wails/v2"
	"github.com/wailsapp/wails/v2/pkg/logger"
	"github.com/wailsapp/wails/v2/pkg/options"
	"github.com/wailsapp/wails/v2/pkg/options/windows"
)

//go:embed frontend/src
var assets embed.FS

//go:embed build/appicon.png
var icon []byte

func main() {
	// Create an instance of the app structure
	app := NewApp()

	m := menu.NewMenuFromItems(
		menu.AppMenu(),
	)

	// Create application with options
	err := wails.Run(&options.App{
		Title:             "sample",
		Width:             1024,
		Height:            768,
		MinWidth:          1024,
		MinHeight:         768,
		MaxWidth:          1280,
		MaxHeight:         800,
		DisableResize:     false,
		Fullscreen:        false,
		Frameless:         false,
		StartHidden:       false,
		HideWindowOnClose: false,
		RGBA:              &options.RGBA{R: 255, G: 255, B: 255, A: 255},
		Assets:            assets,
		Menu:              m,
		Logger:            nil,
		LogLevel:          logger.DEBUG,
		OnStartup:         app.startup,
		OnDomReady:        app.domReady,
		OnBeforeClose:     app.beforeClose,
		OnShutdown:        app.shutdown,
		WindowStartState:  options.Normal,
		Bind: []interface{}{
			app,
		},
		// Windows platform specific options
		Windows: &windows.Options{
			WebviewIsTransparent: false,
			WindowIsTranslucent:  false,
			DisableWindowIcon:    false,
			// DisableFramelessWindowDecorations: false,
			WebviewUserDataPath: "",
		},
		Mac: &mac.Options{
			TitleBar: &mac.TitleBar{
				TitlebarAppearsTransparent: true,
				HideTitle:                  false,
				HideTitleBar:               false,
				FullSizeContent:            false,
				UseToolbar:                 false,
				HideToolbarSeparator:       true,
			},
			Appearance:           mac.NSAppearanceNameDarkAqua,
			WebviewIsTransparent: true,
			WindowIsTranslucent:  true,
			About: &mac.AboutInfo{
				Title:   "Vanilla Template",
				Message: "Part of the Wails projects",
				Icon:    icon,
			},
		},
	})

	if err != nil {
		log.Fatal(err)
	}
}
```

### Expected behaviour

App should launch without problems.

### Screenshots

<img width="2304" alt="Screen Shot 2022-04-18 at 8 39 47 PM" src="https://user-images.githubusercontent.com/1922523/163903054-63580453-135e-41ca-a44f-767c9a15f715.png">

### Attempted Fixes

I have tried different ways of setting a menu but haven't succeeded. It works perfectly, in all ways, on Mac.

### System Details

```shell
Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:             Windows 10 Home
Version:        2009 (Build: 19043)
ID:             21H1
Go Version:     go1.18.1
Platform:       windows
Architecture:   amd64

Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       100.0.1185.44
npm             N/A             Installed       8.5.0
*upx            N/A             Available
*nsis           N/A             Available

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details: 
  - upx : Available at https://upx.github.io/
  - nsis : Available at https://nsis.sourceforge.io/Download
```

### Additional context

I'm running Windows within Parallels Desktop on my Mac

**Comment by leaanthony** [comment]
Ah, Menu Roles are only supported on Mac right now: https://wails.io/docs/next/reference/menus#menuitem. 
You can use the standard functions for menus: 
```go
m := menu.NewMenuFromItems(
		menu.SubMenu("About",
			menu.NewMenuFromItems(
				menu.Text("Hello", nil, nil),
			),
		),
	)
```
Or if you are using master:
```go
	m := menu.NewMenu()
	about := m.AddSubmenu("About")
	about.AddText("Hello", nil, nil)

```

**Comment by selvindev** [comment]
That explains it, thank you👍
